### PR TITLE
feat: use build images to run the application

### PR DESCRIPTION
**Pull request by Yoan-MA**
Instead of building images on the fly, we'll now pulling images from a Container Registry

### CODE REVIEW

File: .github/workflows/deploy-image-on-master-merge.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+      # For each image, build it, tag and push it on registry
+      - name: Build image and push to Registry
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          platforms: linux/amd64, linux/arm/v7, linux/arm64/v8
+          file: ${{ matrix.dockerfile }}
+          push: true
+          tags: |
```
Comment (corenting) [inline .github/workflows/deploy-image-on-master-merge.yaml:74]: We could use https://github.com/docker/metadata-action to generate the tags beforehand in the workflow.

File: .github/workflows/deploy-image-on-master-merge.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      - name: Get current date and time
+        id: date
+        run: echo "date=$(date +'%Y-%m-%d_%H-%M-%S')" >> $GITHUB_OUTPUT
+
+      # For each image, build it, tag and push it on registry
+      - name: Build image and push to Registry
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          platforms: linux/amd64, linux/arm/v7, linux/arm64/v8
```
Comment (corenting) [inline .github/workflows/deploy-image-on-master-merge.yaml:71]: I think we can limit ourselves to linux/amd64 and linux/arm64/v8

File: .github/workflows/deploy-image-on-master-merge.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        id: lowercase
+        run: echo "repository_name_lower_case=${REPOSITORY_NAME,,}" >>$GITHUB_OUTPUT
+        env:
+          REPOSITORY_NAME: '${{ env.REPOSITORY_NAME }}'
+
+      # Use Emulator to build images on different architectures
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
```
Comment (corenting) [inline .github/workflows/deploy-image-on-master-merge.yaml:47]: I don't think we need to specify the argument, it is the default value

**Review by corenting — APPROVED** [review]

File: .github/workflows/deploy-image-on-master-merge.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ steps.lowercase.outputs.repository_name_lower_case }}/${{ matrix.image }}
```
Comment (quentinus95) [inline .github/workflows/deploy-image-on-master-merge.yaml:62]: @Yoan-MA https://github.com/docker/build-push-action/issues/237#issuecomment-848673650

https://github.com/docker/metadata-action#image-name-and-tag-sanitization

> To ease the integration in your workflow, this action will automatically:
> 
> Lowercase the image name

File: docker-compose.yml [hunk 5]
```diff
@@ -2,7 +2,8 @@ version: "3.4"
 
 services:
   listingapi:
-    build: listingapi
+    image: ghcr.io/meilleursagents/backend-technical-test/listingapi:latest
+    user: "listingapi:group_listingapi"
```
Comment (quentinus95) [inline docker-compose.yml:6]: Most of the time it is conventional to have the group matching the user, do you see situations where having the `group_` prefix can be useful?

Comment (Yoan-MA) [reply docker-compose.yml:6]: Nope.

**Pull request merged**